ERCS/erc-7699.md

```diff
@@ -0,0 +1,212 @@
+
+## Motivation
+
+The primary motivation for this proposal is to improve the functionality of the ERC-20 token standard by providing a mechanism for including a payment reference in token transfers, similar to the traditional finance systems where payment references are commonly used to associate and reconcile transactions with specific orders, invoices or other financial records.
+
+Currently, users and merchants who want to include a payment reference in their transactions must rely on off chain external systems or custom payment proxy implementations. In traditional finance systems, payment references are often included in wire transfers and other types of electronic payments, making it easy for users and merchants to manage and reconcile their transactions. Such as:
+
+ - SWIFT MT103: field 70 “Remittance Information” is commonly used for such content (e.g " PAYMENT FOR INVOICE 998877"). There is also field 72 “Sender to receiver information”.
+ - ISO 20022 (for SEPA): PAIN.001 has field called RmtInf (Remittance Information)
+
+By extending the existing ERC-20 token standard with payment transfer reference capabilities, this proposal will help bridge the gap between traditional finance systems and the world of decentralized finance, providing a more seamless experience for users, merchants, and service providers alike.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Any contract complying with ERC-20 when extended with this ERC, MUST implement the following interface:
+```
+// The EIP-165 identifier of this interface is 0x1522573a
+
+interface IERC7699 {
+
+function transfer(address to, uint256 amount, bytes calldata transferReference) external returns (bool);
+
+function transferFrom(address from, address to, uint256 amount, bytes calldata transferReference) external returns (bool);
+
+event TransferReference(bytes32 indexed loggedReference);
+
+}
+```
+
+These `transfer` and `transferFrom` functions, in addition to the standard transfer behaviour, MUST emit a `transferReference` event with a `loggedReference` parameter (with only exception defined below).
+
+The corresponding ERC-20 `Transfer` event MUST be emitted following the `TransferReference` event, ideally immediately afterwards for the client to be able to seek the associated `Transfer` event log record. 
+
+Emitted `loggedReference` MIGHT be the exact copy of the `transferReference` (when less then 33 bytes) or the derived data from the rich `transferReference` structure and other processing. This is up to the implementer. One MUST NOT expect the `transferReference` and `loggedReference` data are equal.
```

Comment:
Is "MIGHT" one of the keywords? I don't actually know.

Reply:
@SamWilsn The keyword is "MAY" . Should I create a separate PR for these fixes , or edit this one?

Reply:
fixed with #716  Thx